### PR TITLE
Extract the new business rule validator to a common project 🔄

### PR DESCRIPTION
**Pull request by kamilbaczek**
## 📋 Description

This refactor is connected with the extraction bussiness rule engine and tests assertion to nuget packages

## 📦 PR Includes

- [ ] Feature added 🆕
- [ ] Bug fix 🐛
- [X] Code refactor 🔄
- [ ] Documentation update 📚
- [ ] Tests added/updated 🧪
- [ ] Other: (describe) 

## 🚫 PR Does Not Include

- [X] Breaking changes ❌
- [X] Major architectural changes 🏗️
- [X] Unrelated features/tasks 🚫

## 💡 Additional Notes

No additional notes

### CODE REVIEW

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/Fitnet.Contracts.Core.csproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
   <ItemGroup>
     <PackageReference Include="ErrorOr" Version="2.0.1" />
-    <PackageReference Include="EvolutionaryArchitecture.Fitnet.Common.Core" Version="4.1.1" />
+    <PackageReference Include="EvolutionaryArchitecture.Fitnet.Common.Core" Version="4.1.0" />
```
Comment (kamilbaczek) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/Fitnet.Contracts.Core.csproj:9]: revert this change

Comment (meaboutsoftware) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/Fitnet.Contracts.Core.csproj:9]: Why not 4.1.2?

**Review by meaboutsoftware — CHANGES_REQUESTED** [review]

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/Fitnet.Contracts.Core.csproj (hunk 1 above)
Comment (kamilbaczek) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/Fitnet.Contracts.Core.csproj]: 4.1.2 version has not been build yet. This will be build after merge

Comment (kamilbaczek) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/Fitnet.Contracts.Core.csproj]: I have to update this in next MR

**Review by meaboutsoftware — APPROVED** [review]
